### DISCUSSION TITLE
How to get a role from a string with the role's id?

### DISCUSSION THREAD

**Discussion by NotAYoutuberYT** [serenity-rs/serenity#2332] [Q&A]
I'm trying to get a role from an ApplicationCommandInteraction's option, but it returns the id as a string. I've tried using Role::convert, but that keeps giving me an error. I've probably missed something very obvious, but I'm tired of looking through documentation and source code. Thanks to anyone who can help!

**Comment by jamesbt365** [comment]
Simply parse the string for the RoleId:
```rs
let role_id = item.parse::<RoleId>();
match role_id {
    Ok(role_id) => role_id,
    // you may want to do something else with the error.
    Err(e) => return Ok(()),
};
```

Then call the `get` method on the `HashMap` of roles on the Guild (I assume you know how to get this from the cache):
https://docs.rs/serenity/latest/serenity/model/guild/struct.Guild.html#structfield.roles

Alternatively, you can do a http request with the `GuildId` instead:
https://docs.rs/serenity/latest/serenity/model/id/struct.GuildId.html#method.roles
